determined-ai/determined
# feat: Input should capture the Esc button and Clicks while focused [WEB-1251]

## Description

This PR introduces the ability for inputs to correctly handle "Escape" button presses and clicks to ensure that the inputs themselves lose focus, instead of having the entire modal close. 

Example showing text, number, and dropdown inputs losing focus and then the modal closing. 


https://github.com/determined-ai/determined/assets/103522725/54683ec0-c9d8-4ba8-87c0-053b53e15d79

**Note:** The functionality handles all inputs in the UI Kit, however there are some input instances throughout the UI, such as some `Select` inputs that do not currently use UI Kit components. The plan is to ensure that there is a follow-up ticket to ensure all inputs use the `UI Kit` components


## Test Plan

1. Navigate to the Workspaces list page.
2. Click on the `New Workspace` button.
3. Click outside the modal, the modal should close.
4.  Click on the `New Workspace` button again.
5. Hit the "Escape" key, the modal should close.
6. Click inside the Workspace name input.
7. Click the escape key. The input should become unfocused and the modal should still be open.
8. Click the escape key again. The modal should now close. 
9. Click on the `New Workspace` button again.
10. Click inside the Workspace name input.
11. Click outside the modal. The input should become unfocused and the modal should still be open.
12. Click outside the modal again. The modal should now close. 
13. Click on the `New Workspace` button again.
14. Click inside the Workspace name input.
15. Click the escape key. The input should become unfocused and the modal should still be open.
16. Click outside the modal. The modal should now close.
17. Click on the `New Workspace` button again.
18. Click inside the Workspace name input.
19. Click outside the modal. The input should become unfocused and the modal should still be open.
20.  Click the escape key . The modal should now close.  
21. Navigate to the Home page.
22. Click the "Launch JupyterLab" button in the top right corner. Repeat steps 3-20 with the "Slots" number input. 
23. Navigate to the "Models" page and click "New Model". Repeat steps 3-20 with the "Description"  textarea input.
24. Click "New Model".  Repeat steps 3-20 with the "Workspace" dropdown. 
25. Click "New Model". Repeat steps 3-20 with the "Description"  textarea input.
26. Click your username in the top left of the page and choose "Settings" in the dropdown. Repeat steps 3-20 with the "Username"  text input.






## Commentary (optional)





## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.

## Ticket

WEB-1251
